src/desktop/apps/articles/test/routes.js

```diff
@@ -0,0 +1,128 @@
+import sinon from 'sinon'
+import articlesJSON from './fixtures.coffee'
+import fixtures from 'desktop/test/helpers/fixtures.coffee'
+import * as _ from 'lodash'
+
+const rewire = require('rewire')('../routes')
+const { articles, section, teamChannel } = rewire
+
+describe('Articles routes', () => {
+  let req
+  let res
+  let next
+  let positronql
+
+  beforeEach(() => {
+    sinon.stub(Backbone, 'sync')
+    req = {
+      params: {},
+      query: {},
+    }
+    res = {
+      render: sinon.stub(),
+      locals: { sd: {} },
+      redirect: sinon.stub(),
+    }
+    next = sinon.stub()
+
+    rewire.__set__(
+      'topParselyArticles',
+      sinon
+        .stub()
+        .yields([
+          fixtures.parselyArticle,
+          fixtures.parselyArticle,
+          fixtures.parselyArticle,
+        ])
+    )
+
+    positronql = sinon.stub().returns(
+      Promise.resolve({
+        articles: articlesJSON,
+      })
+    )
+    rewire.__set__('positronql', positronql)
+  })
+
+  afterEach(() => {
+    Backbone.sync.restore()
+  })
+
+  describe('#articles', () => {
+    it('fetches published articles', () => {
+      articles(req, res, next).then(() => {
+        res.render.args[0][0].should.equal('articles')
+        res.render.args[0][1].articles.length.should.equal(5)
+      })
+    })
+
+    it('requests less than 50 articles', () => {
+      articles(req, res, next).then(() => {
+        positronql.args[0][0].query.should.containEql('limit: 50')
+      })
+    })
+  })
+
+  describe('#section', () => {
+    it('renders the section with articles', () => {
+      const sectionObject = _.extend(_.cloneDeep(fixtures.section), {
+        slug: 'foo',
+      })
+      section(req, res, next)
+      Backbone.sync.args[0][2].success(sectionObject)
+      Backbone.sync.args[1][2].data.section_id.should.equal(sectionObject.id)
+      Backbone.sync.args[1][2].success(fixtures.article)
+      res.render.args[0][0].should.equal('section')
+      res.render.args[0][1].section
+        .get('title')
+        .should.equal(sectionObject.title)
+    })
+
+    it('nexts for an error because it uses a root url', () => {
+      section(req, res, next)
+      Backbone.sync.args[0][2].error()
+      next.called.should.be.ok()
+    })
+  })
+
+  describe('#teamChannel', () => {
+    it('renders the channel with its articles', () => {
+      const channel = _.extend(_.cloneDeep(fixtures.channel), {
+        slug: 'foo',
+        type: 'team',
+      })
+      req.path = 'foo'
+      teamChannel(req, res, next)
+      Backbone.sync.args[0][2].success(channel)
+      Backbone.sync.args[1][2].data.ids.length.should.equal(4)
+      Backbone.sync.args[1][2].success(fixtures.article)
+      res.render.args[0][0].should.equal('team_channel')
+      res.render.args[0][1].channel.get('name').should.equal(channel.name)
+    })
+
+    it('nexts if channel is not a team channel', () => {
+      const channel = _.extend(_.cloneDeep(fixtures.channel), {
```

Comment:
What happens if it is a team channel and gets an error?  Maybe good to add a test for that outcome as well.